test/scheduler_test.rb

```diff
@@ -457,41 +457,41 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
-  test 'gracefully shuts down rufus-scheduler threads' do
+  test 'finishes enqueueing delayed jobs before shutting down' do
     if RUBY_ENGINE == 'jruby' || RUBY_PLATFORM =~ /mingw|windows/i
       omit("forking is not supported on #{RUBY_ENGINE}/#{RUBY_PLATFORM} but " \
            'this behaviour is best tested using forks')
     end
 
-    class BeforeEnqueueJob
-      @queue = :quick
-
-      class << self
-        def before_enqueue_example(*)
-          return false if enqueue_started?
-          enqueue_started!
-
-          sleep 5
-          true
-        end
+    t = Time.now + 1 # in the future
```

Comment:
> The + 1 part is just so that Resque.enqueue_at puts the job in the delayed queue 

what if by the time `enqueue_at` runs, `t` is not smaller than the (then) `Time.now` anymore? I admit that seems unlikely, but possible.